### PR TITLE
Aztec: First Responder Handling

### LINKED ISSUE #6673: Aztec: Title field receives focus upon launch
Mirror first responder behavior from the hybrid editor for the title field.

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details:
- By default, the Title TextField will become the first responder
- Whenever Aztec is about to disappear, we'll store the previous First Responder
- On viewWillAppear, we'll restore the previous first responder
- Did some cleanup in `UIView+Helpers`!

Closes #6673

### To test:
1. Launch Aztec
2. Verify that the Title TextField is the First Responder
3. Press the TextView
4. Open `More` (Top Right navBar button) > `Options`
5. Return to Aztec
6. Verify that the TextView becomes first responder, again.

Needs review: @astralbodies 
cc @diegoreymendez

### CODE REVIEW

**Comment by diegoreymendez** [comment]
I'm taking over this review.  **evil laugh**

File: WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift [hunk 1]
```diff
@@ -290,11 +305,22 @@ class AztecPostViewController: UIViewController {
         startListeningToNotifications()
     }
 
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        // Bring back the keyboard on the last known Responder
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift]: Redundant comment.

File: WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift [hunk 2]
```diff
@@ -290,11 +305,22 @@ class AztecPostViewController: UIViewController {
...
+        restoreFirstResponder()
+    }
+
 
     override func viewWillDisappear(_ animated: Bool) {
         super.viewWillDisappear(animated)
 
+        // Notifications Listeners Cleanup
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift]: Redundant comment.

File: WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift [hunk 3]
```diff
@@ -290,11 +305,22 @@ class AztecPostViewController: UIViewController {
...
         stopListeningToNotifications()
+
+        // Remember the current First Responder
```
Comment (diegoreymendez) [inline WordPress/Classes/ViewRelated/Post/AztecPostViewController.swift]: Redundant comment.

**Comment by jleandroperez** [comment]
@diegoreymendez thanks for the review! ready for another peek!

This mechanism felt *Way* simpler. It was either something like this, or a flags mechanism to... determine who should go up next!

**Comment by diegoreymendez** [comment]
@jleandroperez - Can you cherry pick these changes into `release/7.0` please?

**Comment by jleandroperez** [comment]
@diegoreymendez definitely! Backport-PR submitted in #6704

**Pull request merged**